waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -0,0 +1,82 @@
+module Wasp.Generator.ServerGenerator.JobGenerator
+  ( genJobs,
+    genJobFactories,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import Data.Maybe
+  ( fromJust,
+  )
+import qualified GHC.Enum as Enum
+import StrongPath
+  ( Dir,
+    File',
+    Path,
+    Path',
+    Posix,
+    Rel,
+    parseRelFile,
+    reldir,
+    reldirP,
+    relfile,
+    (</>),
+  )
+import Wasp.AppSpec (AppSpec, getJobs)
+import Wasp.AppSpec.Job (Job (perform))
+import Wasp.Generator.ExternalCodeGenerator.Common (GeneratedExternalCodeDir)
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.JsImport (getJsImportDetailsForExtFnImport)
+import Wasp.Generator.Monad (Generator)
+import Wasp.Generator.ServerGenerator.Common
+  ( ServerSrcDir,
+  )
+import qualified Wasp.Generator.ServerGenerator.Common as C
+
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+data JobFactory = PassthroughJobFactory
+  deriving (Show, Eq, Ord, Enum, Enum.Bounded)
+
+jobFactories :: [JobFactory]
+jobFactories = enumFrom minBound :: [JobFactory]
+
+-- TODO: In future we will detect what type of JobFactory
+-- to use based on what the Job is using.
+jobFactoryForJob :: Job -> JobFactory
+jobFactoryForJob _ = PassthroughJobFactory
+
+jobFactoryFilePath :: JobFactory -> Path' (Rel d) File'
+jobFactoryFilePath PassthroughJobFactory = [relfile|src/jobs/PassthroughJobFactory.js|]
+
+genJobs :: AppSpec -> Generator [FileDraft]
+genJobs spec = return $ genJob <$> getJobs spec
+  where
+    tmplFile = C.asTmplFile [relfile|src/jobs/_job.js|]
+    dstFileFromJobName jobName = C.asServerFile $ [reldir|src/jobs/|] </> fromJust (parseRelFile $ jobName ++ ".js")
+    genJob :: (String, Job) -> FileDraft
+    genJob (jobName, job) =
+      let (jobPerformFnName, jobPerformFnImportStatement) = getJsImportDetailsForExtFnImport relPosixPathFromJobFileToExtSrcDir $ perform job
+       in C.mkTmplFdWithDstAndData
+            tmplFile
+            (dstFileFromJobName jobName)
+            ( Just $
+                object
+                  [ "jobName" .= jobName,
+                    "jobPerformFnName" .= jobPerformFnName,
+                    "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
+                    "jobFactoryName" .= show (jobFactoryForJob job)
+                  ]
+            )
+
+genJobFactories :: Generator [FileDraft]
+genJobFactories = return $ genJobFactory <$> jobFactories
+  where
+    genJobFactory :: JobFactory -> FileDraft
+    genJobFactory jobFactory =
+      let jobFactoryFp = jobFactoryFilePath jobFactory
+          sourceTemplateFp = C.asTmplFile jobFactoryFp
+          destinationServerFp = C.asServerFile jobFactoryFp
+       in C.mkTmplFdWithDstAndData sourceTemplateFp destinationServerFp Nothing
```

Comment:
Since I'm already commenting on a merged PR, I might as well post all that I can think of :smile: 

Following from [these conventions](https://github.com/wasp-lang/wasp/pull/553#discussion_r849560283), I'd probably extract this one to the top-level as well. That said, it's still mostly personal preference. Just putting it out there :)

Reply:
Makes sense, although I have to admit I am still relatively situational about this. If I feel it in some way benefits from the context it gets with `where`, I leave it there. This one? Yeah, this one might be a good candidate for top level, true. Since Job Factory is kind of a standalone concept.

Reply:
Haha yeah these are tricky. On this one, since it was so locally used, I felt where made sense to stay inside the function using it. Hmmm, will think on this one...

Reply:
To quickly further elaborate on my personal preferences here, if I have a function I only intend to use within one other function, but I extracted to make the original function more readable, I typically keep it local (even if it _could_ stand on its own). I guess the question is it preferred to have flat functions with many top-level, or fewer top-level with more local helpers. I am not sure on the Haskell convention here tbh, but using `where` to me it feels more aligned with the "read like a story" argument as well. And subjectively, if I see tons of top-level it becomes harder for me to know what are "meant" to be used (kind of weak, but I hope you get the idea), and what are just one-time use helpers (even if they are below the use) from a quick scanning perspective. But my gut may be way off here 🤷🏻

Reply:
I understand this and agree with it, I mostly think the same way! I guess the subjective think is judging if something is a helper function or not. But ok, at the end I don't think it matters much to be honest, so I would say free for all regarding this.

Reply:
> so I would say free for all regarding this.

Yes, I agree